streams-contrib/streams-processor-urls/src/main/java/org/apache/streams/urls/LinkResolver.java

```diff
@@ -335,72 +271,107 @@ public void unwindLink(String url) {
+        InputStream inputStream = null;
+        try {
+            if (linkDetails.getContentType() != null) {
+                inputStream = connection.getInputStream();
 
-        // Iterate through all the known URL parameters of tracking URLs
-        for (String pattern : URL_TRACKING_TO_REMOVE)
-            url = url.replaceAll(pattern, "");
+                if (inputStream == null) {
+                    linkDetails.setLinkStatus(LinkDetails.LinkStatus.EMPTY);
+                    linkDetails.setRawContent(StringUtils.EMPTY);
+                    return;
+                }
 
-        // If the URL is smaller than when it came in. Then it had tracking information
-        if (url.length() < startLength)
-            linkDetails.setTracked(Boolean.TRUE);
+                if (linkDetails.getContentType().equalsIgnoreCase("text/html")) {
+                    linkDetails.setContentsType(LinkDetails.ContentsType.RAW_HTML);
+                    String rawHTML = convertInputStream(inputStream, connection.getContentEncoding() == null ? "UTF-8" : connection.getContentEncoding().toUpperCase());
 
-        // return our urls.
-        return url;
+                    // we were unable to get the raw information
+                    if (rawHTML == null || rawHTML.equals(StringUtils.EMPTY)) {
+                        linkDetails.setRawContent(StringUtils.EMPTY);
+                        linkDetails.setLinkStatus(LinkDetails.LinkStatus.ERROR);
+                    } else {
+                        linkDetails.setRawContent(rawHTML);
+                        linkDetails.setLinkStatus(LinkDetails.LinkStatus.SUCCESS);
+                    }
+                } else {
+                    linkDetails.setContentsType(LinkDetails.ContentsType.BASE_64);
+                    String base64 = convertInputStreamToBase64(inputStream);
+                    if (base64 == null || base64.equals(StringUtils.EMPTY)) {
+                        linkDetails.setRawContent(StringUtils.EMPTY);
+                        linkDetails.setLinkStatus(LinkDetails.LinkStatus.ERROR);
+                    } else {
+                        linkDetails.setRawContent(base64);
+                        linkDetails.setLinkStatus(LinkDetails.LinkStatus.SUCCESS);
+                    }
+                }
+            }
+        } catch (LinkResolverContentTooLargeException e) {
+            LOGGER.warn("Too Large: {}", connection.getURL().toString());
+            linkDetails.setLinkStatus(LinkDetails.LinkStatus.TOO_LARGE);
+            linkDetails.setRawContent(StringUtils.EMPTY);
+        } catch (SocketTimeoutException e) {
+            LOGGER.warn("Socket Timeout: {}", connection.getURL().toString());
+            linkDetails.setLinkStatus(LinkDetails.LinkStatus.TIME_OUT);
+            linkDetails.setRawContent(StringUtils.EMPTY);
+        } catch (IOException e) {
+            LOGGER.warn("Socket Timeout: {} - {}", e, connection.getURL().toString());
```

Comment:
In these cases where we are logging exceptions, the exception should come last and we shouldn't have to parameterize it in the log string.  This way, we'll get the stack trace for the exception as well.